### ISSUE TITLE
[Bug]: headless: 'new' Chrome crashing instantly issue? (Linux)

### ISSUE THREAD

**Issue by ForbiddenEra** [puppeteer/puppeteer#10265]
### Bug expectation

I expected it to work and not crash?

It crashed, instantly. I can try and grab crash dumps later but I removed the ones that I saw trying to fix things.

I updated my server from Ubuntu 18.04 to 20.04 overnight and then restarted my service; puppeteer didn't want to work at first, figured was because of update, ran `npm update puppeteer` and then started my service and it ran fine, no issues.

Then I saw the deprecation message on the console mentioning new headless mode; read the link, think "neat, maybe this will make things better" and figure I'd give it a quick try.. Added `headless: 'new'` to my launch and tried starting my service.

Service starts, Chrome starts and instantly crashes - I had to adjust the update interval of `htop` to 100ms to even catch that it was starting and disappearing.

That's my main issue; so everything below is kinda secondary, possibly related but not my main concern and might be my own fail, I'll verify this when I can.

No errors. Process just vanishes (in both cases) and then Puppeteer complains that it's gone away.

## **Everything below is secondary/possible user error or separate issue (but _could_ be related) - If you're TL/DR-ish, then you can stop here**.

But then things get _weird._ It's possible that everything below is me missing or messing something; I'll have to try and test and reproduce what follows and ensure it wasn't me going insane or a ghost in the machine.

"Well, crap" I think, maintenance time is almost up and not really the time to debug. Revert the file I added `headless: 'new'` to back to the original file.

Restart the service, Chrome loads, loads the first page (which is a local hosted placeholder page that it will sit on when first launched) and then immediately crashes. I have it setup to stream screenshots to my dashboard, so I was able to see the placeholder load and even the first few frames of animation on the placeholder page (it's basically just a message that constantly grows/shrinks).

"Weird??" I now think.. Try restarting again, same thing. The _only_ difference between it working earlier and now broken is that I added `headless: 'new'` and tried starting a few times and then reverted that change.

Next I try: (note that this is a production service and is now broken after maintenance time and customers are screwed and money is being lost so forgive that the things I try are more brute-force/panicked than nuanced debugging)
- removing and reinstalling Puppeteer
- manually removing Chrome from ~/.cache
- removing any profile stuff from /tmp
- reverting back to the Puppeteer version I was using before updating to Ubuntu 20.04 last night (which was `19.8.2`)
- disabling my screenshot function
- disabling my placeholder page
- trying an even older version of Puppeteer
- manually removing all versions from ~/.cache again
- trying `19.8.2` again
- manually removing all versions from ~/.cache again
- trying `19.8.1`
- manually removing all versions from ~/.cache again
- `npm uninstall puppeteer` and ensuring there's no `puppeteer` or `chrome`-related anything in `./node_modules` or `~/node_modules` or `~/.npm`
- re-installing puppeteer@19.8.2
- manually removing all versions from ~/.cache again
- running `npm update puppeteer` to go current again
- actually rebooting (which for me is almost insane; rebooting Linux to fix something [that's not driver/kernel related at least] is an absolute last resort for me, I'm old-school from the time where having a year of uptime was a good thing, where now-days it just says that you're behind on updates lol plus the mantra of almost nothing on Linux, especially userland, should need a reboot to fix) and trying everything above again

No matter what I tried, I got the same behavior. If I set it back to `headless: 'new'`, instant crash. If I remove it, it crashes but not instantly.

I already have my coworker working on spinning up a new node with last night's pre-update snapshot and I try one last thing: I launch my dev branch of the service (untouched on Puppeteer `19.8.2` and never tried `headless: 'new'`) and it works perfect. 

At this point I'm still very _wtf_, but at least I know that it's launched the same binary from `~/.cache` as the production service was using. I try a few more things, first I tried copying `node_modules` and `package*.json` from dev branch to prod, didn't work.. Then I tried just renaming the whole project directory thinking _maybe_ something somehow is cached based on the directory, no good. Then I move prod to `~/wtf` and copy the dev branch to a new folder where prod was originally. I then copy all source/config files from the original prod in `~/wtf` to the new folder, ensuring it's not using the dev branch config and that none of the changes from the dev branch are included. This works? I've even manually `diff`'d any possibly related file to ensure I didn't miss any possible changes that could've fixed it or caused the issue in the first place.

Maybe I missed something.

Maybe something went SNAFU.

Again, I'm more concerned about `headless: 'new'` crashing in the first place unless someone else tries `headless: 'new'` and gets an instant crash and then removes the `headless: 'new'` and then starts getting a not-quite-instant seemingly different crash, otherwise I'll have to concede that it's some kind of user error on my part and I've missed something.

I'll try later (during off-peak hours) to see if I can reproduce the crash-after-revert on a fresh test branch, otherwise unless it's happened to someone else or anyone can reproduce simply by having it work without `headless: 'new'` then adding `headless: 'new'` and getting an instant crash and then removing `headless: 'new'` and still getting a crash.

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Minimal, reproducible example

```TypeScript
let puppeteer = require('puppeteer');
let browser = await this.puppeteer.launch({
	args: [ '--remote-debugging-port=9222' ]
});
```

### Error string

no error

### Puppeteer version

20.4.0

### Node version

17.9.1

### Package manager

npm

### Package manager version

8.11.0

### Operating system

Linux

**Comment by ForbiddenEra** [comment]
> This issue has an invalid Puppeteer version: `19.8.2, 19.8.1, 20.4.0`. Versions must follow [SemVer](https://semver.org/) formatting. Please update the form with a valid version.
> 
> [Analyzer run](https://github.com/puppeteer/puppeteer/actions/runs/5104949349)

Well - that's **_useful_** when it applies to multiple versions :'(

**Comment by ForbiddenEra** [comment]
> This issue has an outdated Puppeteer version: `19.8.2`. Please verify your issue on the latest `20.4.0` version. Then update the form accordingly.
> 
> [Analyzer run](https://github.com/puppeteer/puppeteer/actions/runs/5104956745)

WoWoW annoying bot.

**Comment by ForbiddenEra** [comment]
> This issue was not reproducible. Please check that your example runs locally and the following:
> 
>     * Ensure the script does not rely on dependencies outside of `puppeteer` and `puppeteer-core`.
> 
>     * Ensure the error string is just the error message.
>       
>       * Bad:
>         ```ts
>         Error: something went wrong
>           at Object.<anonymous> (/Users/username/repository/script.js:2:1)
>           at Module._compile (node:internal/modules/cjs/loader:1159:14)
>           at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
>           at Module.load (node:internal/modules/cjs/loader:1037:32)
>           at Module._load (node:internal/modules/cjs/loader:878:12)
>           at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
>           at node:internal/main/run_main_module:23:47
>         ```
>       * Good: `Error: something went wrong`.
> 
>     * Ensure your configuration file (if applicable) is valid.
> 
>     * If the issue is flaky (does not reproduce all the time), make sure 'Flaky' is checked.
> 
>     * If the issue is not expected to error, make sure to write 'no error'.
> 
> 
> Once the above checks are satisfied, please edit your issue with the changes and we will try to reproduce the bug again.
> 
> [Analyzer run](https://github.com/puppeteer/puppeteer/actions/runs/5104964007)

It might also be prudent to mention that the reproduction code is going to be auto-tested so that people post testable code and not pseudo code or snippets that might use variables defined elsewhere or something...!

**Comment by ForbiddenEra** [comment]
I have a second server spun up from the snapshot that was taken before updating to Ubuntu 20.04; so later on I will try to reproduce both the instant `headless: 'new'` crash and the other issues, both with a minimal test case and with our service before and after updating the second server to Ubuntu 20.04.. should basically be an identical environment since it's a clone, so hopefully I can reproduce everything that happened and hopefully nail down some stuff and hopefully manage to do it with a minimal test case (at least for the main `headless: 'new'` instant-crash issue, the only ppt-related code that ran was what I included above for reproduction)

I'll snag the crash dumps from Chrome too for the `headless: 'new'` instant crash too when I do this; the secondary issue/crash never provided any dumps that I could see.

My brain is still '_wtf_', doubting myself thinking I must've missed a line of code but tracing my steps, doesn't make sense.. literally the only code I changed anyway was adding `headless: 'new'` and then removing it shortly after because it was crashing.

**Comment by OrKoN** [comment]
Could you add the dumpio: true flag to the Puppeteer launch function and post the browser log?

**Comment by OrKoN** [comment]
Thanks for the details! It does sound like this is a bit of out of scope for Puppeteer: perhaps you can file a bug for Chromium at crbug.com? I also wonder if the regular headful chrome crashes when using jemalloc?

**Comment by ForbiddenEra** [comment]
> Thanks for the details! It does sound like this is a bit of out of scope for Puppeteer: perhaps you can file a bug for Chromium at crbug.com? I also wonder if the regular headful chrome crashes when using jemalloc?

Also; since you're probably correct, unless further testing shows that it's somehow related to Puppeteer I'm going to close this issue and if I find the time for more testing I'll open one on crbug.com.

I've actually already tested with the Chrome that Puppeteer installs directly and it does instantly crash with `--headless=new` - so unless the provided Chrome version is specially compiled for Puppeteer than it's very likely unrelated.

I'll have to verify whether previously when it worked (with 19.8.2 or otherwise) that it was indeed using `jemalloc` as if for some reason `jemalloc` wasn't being propagated to Chrome (though I was pretty sure I verified this previously; my entire reason for trying `jemalloc` was to try and eliminate some memory leaks in Chrome/Puppeteer and was pretty sure I had inspected the Chrome process when I initially set it up to ensure Chrome had indeed picked it up) - if I can't prove Chrome previously worked/was using with `jemalloc`, than it's harder to complain about it not working whereas if it did work before and now is broken then it's more reasonable to make a bug/complain :)

I also have to double check that my `libjemalloc.so` wasn't ever manually compiled though I'm also pretty sure I just installed it with `apt` thus it would've/should've been updated when switching to `20.04`; I don't see any left over source files in my home directory indicating I manually compiled it either, but I feel it's only fair to verify all of this now since I did upgrade from `Ubuntu 18.04` -> `Ubuntu 20.04` and hadn't had an issue previously.

Was still definitely a bit of a brainf**k but I was pretty convinced because of the 'weirdness' that it was something I had missed. Actually, when I went to check the enviroment variables it was only because I had seen `/tmp/puppeteer_dev_chrome_profile*` thinking that the `_dev_` part maybe came from a environment variable like `ENV=dev` and then after checking all the vars realized about `jemalloc`!

I still don't understand why my system didn't crash with `jemalloc` in the environment with my system until I set `headless: 'new'` as I had tested it after doing `npm update puppeteer` and it worked fine, again I only knew about `headless: 'new'` because of the console banner warning after starting my service after updating. But considering without `headless: 'new'` it's not an instant crash and considering it's a memory allocator, I suppose anything, especially weird stuff is possible !~
